src/components/ToggleSwitch.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+  <vi-wrapper
+    mini-spacing
+    justify-content="flex-start"
+    class="ViComponent ViToggleSwitch"
+  >
+    <div
+      class="ViToggleSwitch__Container"
+      :class="[
+        {
+          'ViToggleSwitch__Container--mini': mini,
+          'ViToggleSwitch__Container--small': small,
+          'ViToggleSwitch__Container--large': large,
+          'ViToggleSwitch__Container--active': state,
+        }
+      ]"
+      @click.self="onClick">
+      <div
+        class="ViToggleSwitch__Draggable"
+        @mousedown.prevent="dragStart"
+        @click.self="onClick"
+        :style="style">
+      </div>
+    </div>
+  </vi-wrapper>
+</template>
+
+
+<script>
+import ViWrapper from './Wrapper.vue';
+import { scaleMixin } from '../mixins/sizes';
+
+export default {
+  name: 'ViToggleSwitch',
+  components: {
+    ViWrapper,
+  },
+  mixins: [scaleMixin],
+  data() {
+    return {
+      width: 100,
```

Comment:
"DragPercentage" acho que funciona melhor. Isso não é o  a posição do botão no toggle?

Reply:
Não. pos_percentage representa a posição.
width é um fator utilizado para calcular a largura que o pinIn pode se deslocar.

Reply:
Depois com tempo, dá para melhorar até a fórmula, considerando por exemplo o tamanho do pinIn

      let px_position = ((this.position / 100) * this.width)
      let translate_percent = (((px_position / 50)-1)*100)
      return `${(translate_percent < 0) ? 0 : translate_percent}%`